Checks module bodies to be reachable

sobolevn opened the pull request:
This is a very simple change, but now module bodies are also checked to be reachable.
Let's see what effect it will have.

Closes #11331

sobolevn commented on test-data/unit/check-unreachable-code.test:
```diff
@@ -768,13 +768,13 @@ while isinstance(b, int):
-def foo(c: int) -> None:
+def foo(c: int) -> None:  # E: Statement is unreachable
     reveal_type(c)  # N: Revealed type is "builtins.int"
     assert not isinstance(c, int)
```
Nope, because we already have tests for explicit `raise`. Here we test `assert False`

harahu replied:
Ah, I see. I guess my point is that `assert False` isn't a safe way to block off code. But whether mypy recognizes that or not is a more high level design decision, and not a subject for this PR.

hauntsaninja commented on test-data/unit/check-unreachable-code.test:
```diff
@@ -768,13 +768,13 @@ while isinstance(b, int):
     reveal_type(b)  # E: Statement is unreachable
 
-def foo(c: int) -> None:
+def foo(c: int) -> None:  # E: Statement is unreachable
```
Is this resolved? Afaict the `if False` check should be in its own test case right?

sobolevn commented:
@hauntsaninja I forgot to run `git push`  🤦 

```
» git log
commit 2d51fb85261a23242f79e9ffaa4dc95cb7c0c5d0 (HEAD -> issue-11331)
Author: sobolevn <mail@sobolevn.me>
Date:   Fri Oct 22 18:28:03 2021 +0300

    Addresses review
```

hauntsaninja reviewed (approved):
I kind of suspected that :-) Thanks, looks great
